Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -972,68 +970,38 @@ extension MapboxMap {
-        if expected.isError(), let reason = expected.error as? NSString {
+        if expected.isError(), let reason = expected.error {
             throw MapError(coreError: reason)
         }
     }
 
     internal func updateViewAnnotation(withId id: String, options: ViewAnnotationOptions) throws {
         let expected = __map.updateViewAnnotation(forIdentifier: id, options: MapboxCoreMaps.ViewAnnotationOptions(options))
-        if expected.isError(), let reason = expected.error as? NSString {
+        if expected.isError(), let reason = expected.error {
             throw MapError(coreError: reason)
         }
     }
 
     internal func removeViewAnnotation(withId id: String) throws {
         let expected = __map.removeViewAnnotation(forIdentifier: id)
-        if expected.isError(), let reason = expected.error as? NSString {
+        if expected.isError(), let reason = expected.error {
             throw MapError(coreError: reason)
         }
     }
 
     internal func options(forViewAnnotationWithId id: String) throws -> ViewAnnotationOptions {
         let expected = __map.getViewAnnotationOptions(forIdentifier: id)
-        if expected.isError(), let reason = expected.error as? NSString {
+        if expected.isError(), let reason = expected.error {
             throw MapError(coreError: reason)
         }
-        guard let options = expected.value as? MapboxCoreMaps.ViewAnnotationOptions else {
+        guard let options = expected.value else {
             fatalError("Failed to unwrap ViewAnnotationOptions")
         }
         return ViewAnnotationOptions(options)
     }
 
 }
 
-// MARK: - MapProjection
-
-extension MapboxMap {
-    /// Errors related to MapProjection API
-    @_spi(Experimental) public enum MapProjectionError: Error {
-        case unsupportedProjection
-    }
-
-    /// Set map projection for Mapbox map.
-    /// - Parameter mode: The `MapProjection` to be used by the map.
-    /// - Throws: Errors during encoding or `MapProjectionError.unsupportedProjection` if the supplied projection is not compatible with the SDK.
-    @_spi(Experimental) public func setMapProjection(_ mapProjection: MapProjection) throws {
-        try __map.setMapProjectionForProjection(mapProjection.toJSON())
-    }
-
-    /// Get current map projection for Mapbox map.
-    ///
-    /// Please note that even if MapboxMap is configured to use `MapProjection.globe`
-    /// starting from `GlobeMapProjection.transitionZoomLevel` and above
-    /// this method will return `MapProjection.mercator`.
-    ///
-    /// - Returns:
-    ///     `MapProjection` map is using.
-    /// - Throws: Errors during decoding
-    @_spi(Experimental) public func mapProjection() throws -> MapProjection {
-        let data = try JSONSerialization.data(withJSONObject: __map.getMapProjection(), options: [])
-        return try JSONDecoder().decode(MapProjection.self, from: data)
-    }
-}
-
```

Comment:
Removing instead of deprecating in keeping with Android. https://github.com/mapbox/mapbox-maps-android/pull/1255#discussion_r842709350